process/smartContract/process.go

```diff
@@ -230,9 +233,21 @@ func (sc *scProcessor) doExecuteSmartContractTransaction(
 		return vmcommon.UserError, sc.ProcessIfError(acntSnd, txHash, tx, err.Error(), []byte(returnMessage), snapshot)
 	}
 
+	err = sc.checkUpgradePermission(acntSnd, acntDst, vmInput)
+	if err != nil {
+		log.Debug("checkUpgradePermission", "error", err.Error())
+		return vmcommon.UserError, sc.ProcessIfError(acntSnd, txHash, tx, err.Error(), []byte(err.Error()), snapshot)
+	}
+
 	executedBuiltIn, err = sc.resolveBuiltInFunctions(txHash, tx, acntSnd, acntDst, vmInput)
 	if err != nil {
-		log.Debug("processed built in functions error", "error", err.Error())
+		// QUESTION FOR REVIEW: before, this was log.Debug() and, in our tests, resulted in a lot of:
+		// "processed built in functions error error = insufficient funds",
+		// Is it OK to have this new condition on log.Debug() here?
+		if !executedBuiltIn {
+			log.Debug("processed built in functions error", "error", err.Error())
```

Comment:
move to log.Trace - no need for log .debug

Reply:
Fixed.